x/nft/codec.proto

```diff
@@ -32,10 +32,23 @@ message AddApprovalMsg {
     bytes address = 2;
     string action = 3;
     ApprovalOptions options = 4 [(gogoproto.nullable) = false];
+    Type t = 5;
 }
 
 message RemoveApprovalMsg {
     bytes id = 1;
     bytes address = 2;
     string action = 3;
+    Type t = 4;
+}
+
+enum Type {
```

Comment:
If I want to add another nft in a subdir, I have to update this file in parent, right?

Reply:
pp: for simple extensibility we would need a string type like actions.

Reply:
@alpe  all the nft types are to be defined here, so I'm not sure why would we need string ops here? I like the fact that this automatically validates things, whereas we have to validate strings ourselves.

Reply:
This is a bigger discussion and I would but it in an issue/pr outside of this one.
Protobuf is nice in that it gives you fixed types, but it does make it a bit tricky to allow one to extend it.

I'd love a brainstorm on how to do this cleanly, but as I mentioned, changing this part is not an issue for this pr.